src/typegraph/specs/codegen/rpc/typescript/client.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+import fs from "node:fs";
+import { Buffer } from "node:buffer";
+
+const BUFFER_SIZE = 1024;
+
+const state = { id: 0 };
+const isDeno = !Deno.version.v8.includes("node");
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+type RpcResponse<R, E = null> = {
+  jsonrpc: "2.0";
+  result?: R;
+  error?: {
+    code: number;
+    message: string;
+    data?: E;
+  };
+  id: number | string;
+};
+
+function toCamelCase(str: string) {
+  return str.replace(/_([a-z])/g, (_, letter) => letter.toUpperCase());
+}
+
+function toSnakeCase(str: string) {
+  return str.replace(/([A-Z])/g, "_$1").toLowerCase();
+}
+
+function transformKeys(obj: any, convertKey: (key: string) => string): any {
+  if (Array.isArray(obj)) {
+    return obj.map((item) => transformKeys(item, convertKey));
+  } else if (obj && typeof obj === "object") {
+    const result: Record<string, any> = {};
+
+    for (const [key, value] of Object.entries(obj)) {
+      const newKey = convertKey(key);
+      result[newKey] = transformKeys(value, convertKey);
+    }
+
+    return result;
+  }
+
+  return obj;
+}
+
+function readResponse() {
+  const buffer = Buffer.alloc(BUFFER_SIZE);
+
+  let bytesRead = null;
+  let content = Buffer.alloc(0);
+
+  if (isDeno) {
+    do {
+      bytesRead = Deno.stdin.readSync(buffer) ?? 0;
```

Comment:
Issues with the `node:fs` impl on Deno?

Reply:
Actually there's an inconsistency with node:fs readSync behavior so I decided to separate them